### PR TITLE
feat: new fast core documentation

### PR DESCRIPTION
**Pull request by joelkeener**

### CODE REVIEW

File: docs/API.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+    .then((template) => {
+        console.log(template.getParametersSchema());
+        console.log(template.render({
+            var: "value",
+            boolVar: false
+        }));
+    });
+```
+
+Note that despite loading a single template set, a template set name must still be provided when querying the provider.
```
Comment (F5Debbie) [inline docs/API.md]: Please bold the Note and drop "that"  
Format as:
**Note:** Despite loading a single...
Double ** on each side of a word/words will bold them.

File: docs/Advanced.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+const templateSetPath = '/path/to/templateSet';
+const templateProvider = new fast.FsSingleTemplateProvider(templateSetPath);
+
+templateProvider.getSetData('templateSetName')
+    .then((tsData) => {
+        console.log(tsData.hash)
+    });
+```
+
+#### Note
```
Comment (F5Debbie) [inline docs/Advanced.md]: Please put the Note: and the sentence on the same line.

**Note:** Despite loading a single template...

File: docs/GUI.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
+## Schema-driven GUI
+
+Generating a GUI is done by passing data to a libary called JSON Editor. JSON supports formats for schema of type string.
+
+Example of a password object.
```
Comment (F5Debbie) [inline docs/GUI.md]: Change the end period to a colon.

Example of a password object:

File: docs/GUI.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "properties": {
+    "password": {
+      "type": "string",
+      "format": "password"
+    }
+  }
+}
+```
+
+#### Note
```
Comment (F5Debbie) [inline docs/GUI.md]: Combine lines 18 and 19.

**Note:** If property enum is specified, type format is ignored.

File: docs/QuickStart.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+### ContentTypes
+
+When BIG-IP FAST renders it is doing string replacement via Mustache, which is agnostic to the output type. However, specifying a contentType in the template can enable some additional features:
+
+* Post-processing steps (e.g., strip dangling commas for JSON)
+* Smarter merges
+* Smarter handling of some data types
+
+#### Note:
```
Comment (F5Debbie) [inline docs/QuickStart.md]: Start the Note: on line 57

File: docs/SubTemplating.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+2. 2[0-4][0-9] - Matches numbers between 200 and 249, where the first digit is always 2, the second is between 0 and 4, and the third digit is any between 0 and 9,
+3. 25[0-5]: (the 3rd group) matches numbers between 250 and 255, where 25 is always at front and the third digit is between 0 and 5.
+
+#### See also
+JSON schema [Built-in Formats](https://json-schema.org/understanding-json-schema/reference/string.html?highlight=maxlength#built-in-formats) and [Regular Expressions](https://json-schema.org/understanding-json-schema/reference/string.html#id6) for more information.
+
+**Boolean:** The boolean type { type: boolean } matches two values; true or false and must be used in all lower case characters.
+
+
+# Combining Schema¶
```
Comment (F5Debbie) [inline docs/SubTemplating.md]: Remove the paragraph marker.

File: docs/API.md [hunk 7]
```diff
@@ -0,0 +1,197 @@
+
+## Module API
```
Comment (mstokes-f5) [inline docs/API.md]: Since this is now it's own page, we should drop a level of headings (use `#` instead of `##`, and so on for all sub-sections).

File: docs/Advanced.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
+# Advanced Features
+
```
Comment (mstokes-f5) [inline docs/Advanced.md]: We should add an overview of what this page is about: a grab-bag of more advanced topics that did not fit nicely elsewhere.

File: docs/Advanced.md [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+    template: |
+        {{var}}
+`;
+
+fast.Template.loadYaml(yamldata)
+    .then(template => template.forwardHttp()); // POST "foo" to http://example.com/resource
+```
+
+## Multiple Choice Lists
+While authoring a template, it is possible to specify multiple choice list boxes for use in cases such as iRules.
```
Comment (mstokes-f5) [inline docs/Advanced.md]: We want to make these docs BIG-IP-agnostic, so we should chose a different example than iRules.

Actually, upon further reflection, I believe this whole section (Multiple Choice Lists) is better suited for FAST on BIG-IP docs (i.e., Cloud Docs) since:

* the code example relies on the `enumFromBigip` BIG-IP extension
* this is about how to generate a UI from schema, which is specific to the tools/library used (e.g., JSON Editor used by FAST for BIG-IP).

File: docs/GUI.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
+## Schema-driven GUI
```
Comment (mstokes-f5) [inline docs/GUI.md]: I wonder if we leave this page on Cloud Docs for now since it is pretty specific to BIG-IP and JSON Editor. We could add a new page later to discuss generating GUIs in a more generic sense. We could go into some details about how JSON Editor is well tested for this use-case and also that we provide some utilities that make it easier to use JSON Editor with FAST-generated schema.

File: docs/QuickStart.md [hunk 11]
```diff
@@ -0,0 +1,58 @@
+## Getting Started / Quick Start
+
+### Features
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: A features list seems better suited for a README

File: docs/QuickStart.md [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+* Parses Mustache templates and an extended template format (in YAML)
+* Generates a view schema from parsed template data
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: Use "parameter schema" instead of "view schema" for consistency

File: docs/QuickStart.md [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+* Supports Mustache partials and sections
+* Renders templates with user-provided views
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: "user-provided parameters"

File: docs/QuickStart.md [hunk 14]
```diff
@@ -0,0 +1,58 @@
...
+* Validates user-provided views against generated view schema
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: "view(s)" -> "parameter(s)"

File: docs/QuickStart.md [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+* Includes a [command line interface](#cli)
+
+### CLI
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: Basic CLI usage may be better suited for the README

File: docs/QuickStart.md [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+The CLI can also be accessed by executing `cli.js`.
+For example:
+
+```bash
+./cli.js render path/to/template
+```
+
+### ContentTypes
+
+When BIG-IP FAST renders it is doing string replacement via Mustache, which is agnostic to the output type. However, specifying a contentType in the template can enable some additional features:
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: @F5Debbie, should we be using "BIG-IP FAST" when referring to the SDK/language/core, or just when we are talking about the FAST product that runs on BIG-IP? I would prefer that these docs do not even mention BIG-IP to help with the messaging that `f5-fast-core` and the template language are agnostic to how the rendered output is used.

File: docs/QuickStart.md [hunk 17]
```diff
@@ -0,0 +1,58 @@
...
+fast <command> --help
+```
+
+The CLI can also be accessed by executing `cli.js`.
+For example:
+
+```bash
+./cli.js render path/to/template
+```
+
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: For a getting started, we should add an example with a basic template and show rendering it via the CLI.

File: docs/QuickStart.md [hunk 18]
```diff
@@ -0,0 +1,58 @@
...
+### ContentTypes
+
+When BIG-IP FAST renders it is doing string replacement via Mustache, which is agnostic to the output type. However, specifying a contentType in the template can enable some additional features:
+
+* Post-processing steps (e.g., strip dangling commas for JSON)
+* Smarter merges
+* Smarter handling of some data types
+
+#### Note:
+Since BIG-IP FAST relies on BIG-IP AS3 for templating, a contentType of application/json should be declared indicating that the body format is JSON. See [What is the correct JSON Content Type](https://stackoverflow.com/questions/477816/what-is-the-correct-json-content-type) for additional information.
```
Comment (mstokes-f5) [inline docs/QuickStart.md]: This note seems more appropriate for Cloud Docs rather than this set of docs.

**Review by mstokes-f5 — CHANGES_REQUESTED** [review]
I have been focusing more on the general content rather than any specific wording or code examples. Most of this information is coming from sources that have already undergone a more thorough review anyways.

I also still need to review SubTemplating.md and Templating.md, but I wanted to get some feedback out.

We should also edit the README to remove the now duplicate information and link to the appropriate doc files.

I would prefer all lowercase for the filenames to better match the filenames in the rest of the project.

File: docs/Advanced.md (hunk 8 above)
Comment (joelkeener) [reply docs/Advanced.md]: How's this:
    This page offers more advanced features that do not appear in other related documentation for FAST Core.
?

File: docs/GUI.md (hunk 10 above)
Comment (joelkeener) [reply docs/GUI.md]: Would you like to leave it here until we get that more generic, JSON Editor documentation, or is it better to just remove this page until then?

File: docs/QuickStart.md (hunk 12 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I've updated this in the README #features.

File: docs/QuickStart.md (hunk 13 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I've updated this in the README #features.

File: docs/QuickStart.md (hunk 14 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I've updated this in the README #features.

File: docs/QuickStart.md (hunk 15 above)
Comment (joelkeener) [reply docs/QuickStart.md]: Would the link to the CLI section of the Quick Start be sufficient on the README, 
or do we need to either duplicate this info on both pages or 
remove the CLI section from the Quick Start and only have it on the README?

File: docs/QuickStart.md (hunk 18 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I have removed this note.

File: docs/API.md (hunk 1 above)
Comment (joelkeener) [reply docs/API.md]: I have done this, but with a blockquote, instead of an h4, so that the note could be bold and the rest of the line would not.
I have also changed all the See Also notes to match this format.

File: docs/API.md (hunk 7 above)
Comment (joelkeener) [reply docs/API.md]: This is done for all pages.

File: docs/Advanced.md (hunk 9 above)
Comment (joelkeener) [reply docs/Advanced.md]: This has been removed.

File: docs/QuickStart.md (hunk 11 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I have moved these to the README

File: docs/QuickStart.md (hunk 16 above)
Comment (F5Debbie) [reply docs/QuickStart.md]: Just for products running on BIG-IP.  No need for language etc unless we hear otherwise.

File: docs/QuickStart.md (hunk 17 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I added the simple template loading example from the API documentation to the Quick Start to show rendering it with the cli.

File: docs/QuickStart.md (hunk 16 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I have changed BIG-IP FAST to FAST in subtemplating.md (3 places) and in templating.md (7 places) but there are other references to BIG-IP FAST, or just BIG-IP, that I did not remove as it references BIG-IP / BIG-IP FAST, not the FAST Core.

File: docs/Advanced.md (hunk 8 above)
Comment (mstokes-f5) [reply docs/Advanced.md]: Works for me

File: docs/QuickStart.md (hunk 16 above)
Comment (joelkeener) [reply docs/QuickStart.md]: Guillermo has reservations about doing this, and will discuss them with Mitchell

File: docs/GUI.md (hunk 10 above)
Comment (joelkeener) [reply docs/GUI.md]: I have removed this for now.

File: docs/QuickStart.md (hunk 15 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I have added the TOC to a README file in the docs/ dir (README is the only file name GitHub will render for a folder).

File: docs/QuickStart.md (hunk 16 above)
Comment (joelkeener) [reply docs/QuickStart.md]: I reverted the commit that changed BIG-IP FAST to FAST where referencing FAST Core.

File: docs/templating.md [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+  {
+    "data": {{base64var}}
+  }
+```
+
+> **See Also:** [AS3 Schema Reference](https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/refguide/schema-reference.html) for a full list of f5base64 fields.
+
+
+# Enumerated Type
+An Enumerated Type (enum) is a list of constant values. In order for the property to be valid, it must match an item in the list of values. As it relates to BIG-IP, it is a mechanism to pull data from the BIG-IP (enumFromBigip) presenting it as an enum.
```
Comment (mstokes-f5) [inline docs/templating.md]: `enumFromBigip` is a BIG-IP FAST extension and should not be documented her -- it should be on Cloud Docs.

File: docs/templating.md [hunk 20]
```diff
@@ -0,0 +1,188 @@
...
+        toBase64: true
+    template: |
+        {{var}}
+```
+
+Similarly, if the data file is base64-encoded, it can be decoded using `fromBase64`.
+If both `toBase64` and `fromBase64` are set, then `toBase64` takes precedence.
+
+# Base64 Encoding
+Using the GUI, BIG-IP FAST has the ability to encode template parameters as base64, which becomes part of the template output (AS3 declaration).
```
Comment (mstokes-f5) [inline docs/templating.md]: We shouldn't be discussing the BIG-IP FAST GUI as part of Core docs. We also should not be assuming that the rendered output is an AS3 declaration.

We should be able to explain this feature without reaching for BIG-IPisms: It takes a string parameter and base64 encodes it before placing it in the rendered output.

File: docs/templating.md [hunk 21]
```diff
@@ -0,0 +1,188 @@
+# Templating with F5 BIG-IP FAST
+
+This chapter is dedicated to explaining the relationship of schema vs templates. BIG-IP FAST makes use of Mustache, JSON schema and JSONPath, therefore BIG-IP FAST may be familiar if you already understand any of these syntaxes. BIG-IP FAST combines these technologies to provide a complete templating solution. At the foundation, BIG-IP FAST uses a templating specification called mustache to convert parameters into a fully rendered API body. Parameter types can be specified in JSON schema, and are used to validate template inputs when the template is rendered. BIG-IP FAST will auto generate a schema for each template based off the template and json-schema provided. Schema is generated from template text, combined with definitions, and used to validate template parameters.
```
Comment (mstokes-f5) [inline docs/templating.md]: For version control (and code review), it is better to have each sentence on its own line. They will form a single paragraph when rendered (a blank line is needed for a paragraph break). As an example, the section on `Sections` does this.

I feel like this page does not explicitly cover "the relationship of schema vs templates," and it is far from "dedicated to [it]." Instead, we could opt for simpler, but still descriptive language: This chapter (page? section?) goes into the details of templating with BIG-IP FAST. We could also just drop the line and dive in.

The lines that follow that discuss the various technologies in FAST seems awkwardly worded and contain some weirdness like "convert parameters into a fully rendered API body." Instead of trying to pick apart individual sentences and phrases, I will just provide an example we can work off of, and we can chat about specifics if you'd like:

BIG-IP FAST leverages existing, familiar technologies such as Mustache and JSON Schema to provide a complete templating solution that supports comprehensive parameter validation. Template text is written in Mustache (add link). That template text is then parsed to extract any parameters and provide basic JSON Schema for them. Users can write their own JSON Schema for each parameter to expand on the basic schema auto-generated by BIG-IP FAST. When rendering templates, this combined schema is then used to validate any inputs (parameters) to the template.

File: docs/templating.md [hunk 22]
```diff
@@ -0,0 +1,188 @@
...
+
+## Mustache
```
Comment (mstokes-f5) [inline docs/templating.md:9]: This whole section (and I include the Tags, Sections, etc sections in this) seems pretty redundant with Mustache docs, and duplicating a lot of that information here just opens us up for small errors or verbiage that seems not quite right (e.g., I am unsure about the engine vs spec stuff).

Instead, I believe we should link to Mustache docs and then give some use-case-driven examples of how to use Mustache to write templates. For example, we could show:

* Basic string replacement
* Lists
* Conditionals
* Maybe something around partials

File: docs/templating.md [hunk 23]
```diff
@@ -0,0 +1,188 @@
...
+iRules are a common use case, however AS3 supports base64 for a wide range of objects.
+
+In the following example, base64var will display as editable plain text but render as base64-encoded text:
+```yaml
+contentType: application/json
+definitions:
+    base64var:
+        type: string
+        format: text
+        contentEncoding: base64
```
Comment (mstokes-f5) [inline docs/templating.md]: This might be a JSON Editor thing. Instead, we should discuss using `toBase64` and `fromBase64`.

File: docs/templating.md [hunk 24]
```diff
@@ -0,0 +1,188 @@
...
+```yaml
+definitions:
+    var:
+        dataFile: example
+        toBase64: true
+    template: |
+        {{var}}
+```
+
+Similarly, if the data file is base64-encoded, it can be decoded using `fromBase64`.
```
Comment (mstokes-f5) [inline docs/templating.md:198]: Should data files come after the Base64 Encoding section?

File: docs/template_merging.md [hunk 25]
```diff
@@ -0,0 +1,147 @@
+# Overlaid Definitions
```
Comment (mstokes-f5) [inline docs/template_merging.md:1]: How parameter definitions/schema is generated should be part of templating, and not template_merging.

File: docs/template_merging.md [hunk 26]
```diff
@@ -0,0 +1,147 @@
...
+
+The way BIG-IP FAST generates parameter definitions can be surprising at times if that parameter shows up multiple times in the template text.
```
Comment (mstokes-f5) [inline docs/template_merging.md:3]: It really should not be "surprising." We can just dive in an explain it. In other words, delete this line and just rely on the second.

File: docs/template_merging.md [hunk 27]
```diff
@@ -0,0 +1,147 @@
...
+
+When generating parameter definitions, BIG-IP FAST looks at the following locations in the following order, with later definitions overriding/modifying previous ones:
+1. Embedded mustache tags in any merged templates.  For example: {{var:f5:port}}
+2. The definitions properties of any merged templates. Templates are merged by name using $ref inside a oneOf, anyOf, or allOf clause.
```
Comment (mstokes-f5) [inline docs/template_merging.md:7]: Instead of describing how templates are merged as part of this list, we can just add a note like:
... of any merged templates (see here[link] for information on template merging)

File: docs/template_merging.md [hunk 28]
```diff
@@ -0,0 +1,147 @@
+# Overlaid Definitions
+
+The way BIG-IP FAST generates parameter definitions can be surprising at times if that parameter shows up multiple times in the template text.
+
+When generating parameter definitions, BIG-IP FAST looks at the following locations in the following order, with later definitions overriding/modifying previous ones:
+1. Embedded mustache tags in any merged templates.  For example: {{var:f5:port}}
```
Comment (mstokes-f5) [inline docs/template_merging.md:6]: This is minor, but do we really need to end list items with periods? The couple of cases where we have a second sentence, we could just use parenthesis. For example:

... in any merged templates (e.g., `{{var:f5:port}}`)

File: docs/template_merging.md (hunk 27 above)
Comment (mstokes-f5) [inline docs/template_merging.md:7]: Let's make keys/properties of template files stand out a bit by marking them as code. For example:

The `definitions` properties of ...

File: docs/template_merging.md [hunk 29]
```diff
@@ -0,0 +1,147 @@
...
+2. The definitions properties of any merged templates. Templates are merged by name using $ref inside a oneOf, anyOf, or allOf clause.
+3. Embedded mustache tags in the primary template.
+4. The definitions property in the primary template.
+5. The parameters property in any merged templates.
+6. The parameters property in the primary template.
+
+#### Notes:
```
Comment (mstokes-f5) [inline docs/template_merging.md:13]: We should not use section/header-level control for styling. Maybe just bold/emphasis it?

File: docs/template_merging.md [hunk 30]
```diff
@@ -0,0 +1,147 @@
...
+* If a duplicate Mustache tag exists in the template, then the last encountered tag is used for the definition. The order that Mustache tags are parsed in should not be assumed.
+* Properties within the definition (e.g., title, description, type, format, default, etc.) are merged together as they are found with newer data taking precedence over old data on key conflicts.
+* Values from the parameters property of YAML templates will be used in place of the default from the parameter definition but will not actually update the definition itself.
+
+
+# JSON Schema Basic Types
```
Comment (mstokes-f5) [inline docs/template_merging.md:19]: This also feels like something that belongs in templating and template_merging

File: docs/template_merging.md [hunk 31]
```diff
@@ -0,0 +1,147 @@
...
+
+## Definitions
+JSON Schema allows us to define auxiliary schema in order to be reused and combined later on.
+This involves two steps:
+1. We need to define the subschemas to be used later on, and
+2. We need a standard for calling and reusing these definitions.
+
+To establish a difference between the main schema and the auxiliary definitions, we adopt the convention that every JSON Schema document consists of two parts; a JSON Schema, and a set of definitions.
```
Comment (mstokes-f5) [inline docs/template_merging.md]: This talk of main/auxiliary schema/definitions seems confusing to me. I am also not sure what the line about that adopted convention is even trying to say.

I suggest dropping all of this block of text, and just going with something like:

Users can control the JSON Schema FAST generates for parameters by providing parameter "definitions."
These definitions are added to the top-level `definitions` property of a template file.

File: docs/template_merging.md [hunk 32]
```diff
@@ -0,0 +1,147 @@
...
+
+For example, if we want a definition for virtuals, it may look like this:
```
Comment (mstokes-f5) [inline docs/template_merging.md]: Instead of using "virtuals" (a BIG-IPism) for examples, we could just use a list of IP addresses (`ipaddrs`?)

File: docs/template_merging.md [hunk 33]
```diff
@@ -0,0 +1,147 @@
...
+        items:  {
+                type: string,
+                format: ipv4
+            }
+```
+
+> **See Also:** [JSON Editor: $ref and definitions](https://github.com/json-editor/json-editor#ref-and-definitions) for additional code examples.
+
+**Array:** Arrays are used for ordered elements.
+In JSON, each element in an array may be of a different type. Elements of the array may be ordered or unordered based on the API being templated. This section covers typical JSON schema definitions for common patterns.
```
Comment (mstokes-f5) [inline docs/template_merging.md]: FAST is not just for templating APIs. This also mentions covering typical definitions for common patterns, but then it only shows one example.

File: docs/template_merging.md [hunk 34]
```diff
@@ -0,0 +1,147 @@
...
+```yaml
+parameters:
+    ...
+definitions:
+    ...
+template: |
+    ...
+allOf:
+    - $ref: "tcp.yaml#"
+```
```
Comment (mstokes-f5) [inline docs/template_merging.md]: This seems to be the only thing that actually discusses template merging in this page. We should start with this and expand on it.

File: docs/templating.md (hunk 19 above)
Comment (joelkeener) [reply docs/templating.md]: So shall I just remove this explanation about how it is used in relation to BIG-IP:
`As it relates to BIG-IP, it is a mechanism to pull data from the BIG-IP (enumFromBigip) presenting it as an enum.`
?

Or should we remove the whole section?

File: docs/templating.md (hunk 20 above)
Comment (joelkeener) [reply docs/templating.md]: Should we remove this section, or just reword it to not mention the GUI or BIG-IP?

File: docs/templating.md (hunk 21 above)
Comment (joelkeener) [reply docs/templating.md]: What do you think of this, with only one update (added the words "that is" before "auto-generated by BIG-IP FAST"):

BIG-IP FAST leverages existing, familiar technologies such as Mustache and JSON Schema to provide a complete templating solution that supports comprehensive parameter validation. Template text is written in Mustache (add link). That template text is then parsed to extract any parameters and provide basic JSON Schema for them. Users can write their own JSON Schema for each parameter to expand on the basic schema that is auto-generated by BIG-IP FAST. When rendering templates, this combined schema is then used to validate any inputs (parameters) to the template.

File: docs/templating.md (hunk 19 above)
Comment (mstokes-f5) [reply docs/templating.md]: I think we can just remove the whole section for now.

File: docs/templating.md (hunk 21 above)
Comment (mstokes-f5) [reply docs/templating.md]: Sounds good to me.

File: docs/templating.md (hunk 20 above)
Comment (mstokes-f5) [reply docs/templating.md]: I took another look at our base64 features, and I now believe this section is actually specific to BIG-IP FAST. The `toBase64` and `fromBase64` in the previous sections are only for data files. In other words, we should remove this section.

Comment (joelkeener) [reply docs/templating.md]: I think we should remove the whole section.

File: docs/templating.md (hunk 19 above)
Comment (joelkeener) [reply docs/templating.md]: I have just removed the explanation about how the enum relates to BIG-IP

Comment (joelkeener) [reply docs/templating.md]: I see that you want to remove the whole section though, so I have that done now.

File: docs/template_merging.md (hunk 33 above)
Comment (joelkeener) [reply docs/template_merging.md]: What is another example of a a typical definitions for common patterns that we can use?

File: docs/template_merging.md (hunk 34 above)
Comment (joelkeener) [reply docs/template_merging.md]: With the updates to the first confusing example on the template_merging.md page, we now discuss a more simple example of top-level definition merging first, and then this example -- which should probably still be expanded upon, but I have not done that. 

Should we leave this at the end and expand on it for more specific schema merging options?

File: docs/template_merging.md (hunk 32 above)
Comment (joelkeener) [reply docs/template_merging.md]: I have made this update, replacing virtuals with ip_addrs.

File: docs/template_merging.md (hunk 31 above)
Comment (joelkeener) [reply docs/template_merging.md]: I have replaced the confusing verbiage with your simpler explanation, and added a better example to clarify its meaning.

File: docs/template_merging.md (hunk 30 above)
Comment (joelkeener) [reply docs/template_merging.md:19]: Should I add a copy of this in the templating.md file somewhere?

If so, do you mean the JSON Schema Basic Types or the bullet point before it or both?

File: docs/templating.md (hunk 23 above)
Comment (joelkeener) [reply docs/templating.md]: toBase64 and fromBase64 are discussed at the end of this file, in the Template Data Files section. 
Should that be moved to the Base64 section -- which I have removed since it is specific to JSON Editor, while to/from base64 is data specific and might only belong where it is now?

File: docs/templating.md (hunk 22 above)
Comment (joelkeener) [reply docs/templating.md:9]: Please review the links, examples and simplified text that I've updated in the following sections of docs/templating.md:
- Mustache
- Sections
- Tags

File: docs/templating.md (hunk 20 above)
Comment (joelkeener) [reply docs/templating.md]: The Base64 Encoding was removed

File: docs/template_merging.md (hunk 33 above)
Comment (elliotwilcock-cli) [reply docs/template_merging.md]: Moved to a new JIRA item.

File: docs/template_merging.md (hunk 30 above)
Comment (elliotwilcock-cli) [reply docs/template_merging.md:19]: Moved to a new JIRA item
